### PR TITLE
[PT FE] Remove warnings suppression in frontend code

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Remove warnings suppression in frontend code*

### Tickets:
 - *103431*

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

File: src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp [hunk 1]
```diff
@@ -62,11 +62,12 @@ AppendListUnpackReplacer::AppendListUnpackReplacer() {
...
-            if (index_val[0] >= 0) {
-                index = index_val[0];
-            } else {
-                index = inputs.size() + index_val[0];
+            if (index_val.size() != 1) {
+                return false;
+            }
+            auto index = index_val[0];
+            if (index_val[0] < 0) {
+                index = inputs.size() + index;
```
Comment (rkazants) [inline src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp:70]: here we try to sum-up `size_t` and `int64_t` element. Please check

File: src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp [hunk 2]
```diff
@@ -93,15 +93,16 @@ AtenGetItemReplacer::AtenGetItemReplacer() {
...
-                    index = index_val[0];
-                } else {
-                    index = split->outputs().size() + index_val[0];
+                auto index_val = getitem_index_const->cast_vector<int64_t>();
+                if (index_val.size() != 1) {
+                    return false;
+                }
+                auto index = index_val[0];
+                if (index < 0) {
+                    index = split->outputs().size() + index;
```
Comment (rkazants) [inline src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp:105]: Please also check

File: src/frontends/pytorch/src/utils.cpp [hunk 3]
```diff
@@ -232,7 +232,7 @@ OutputVector make_framework_node(NodeContext* context) {
                                           context->get_output_size() - num_body_outs + num_skip_body_outputs);
     fw_node->set_friendly_name(context->get_op_type());
     for (size_t i = 0; i < bodies.size(); ++i) {
-        fw_node->set_function(i, bodies[i]);
+        fw_node->set_function(static_cast<int>(i), bodies[i]);
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.cpp:235]: may be it is a problem of API `set_function`? Why do we use `int` type for body index. `size_t` is better.

File: src/frontends/pytorch/src/utils.hpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ void num_inputs_check(const NodeContext& context, size_t min_inputs, size_t max_
 
 Output<Node> make_optional_bias(const Output<Node>& base_op,
                                 const NodeContext& context,
-                                size_t bias_input_idx,
+                                int bias_input_idx,
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.hpp:25]: `size_t` is more logical.

**Comment by rkazants** [comment]
@ilya-lavrenov, please take a look at this PR

File: src/frontends/pytorch/src/utils.hpp (hunk 4 above)
Comment (mvafin) [reply src/frontends/pytorch/src/utils.hpp:25]: It is used for API that uses `int` as input, so same type makes sense.

File: src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp (hunk 1 above)
Comment (mvafin) [reply src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp:70]: If compiler doesn't output warnings I think it is cleaner to not use extra casts, @ilya-lavrenov what do you think?

File: src/frontends/pytorch/src/utils.cpp (hunk 3 above)
Comment (rkazants) [reply src/frontends/pytorch/src/utils.cpp:235]: if it is not FE code, I am fine then. I thought this is PT FE FW Node API:)

File: src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/pytorch/src/transforms/append_list_unpack_replacer.cpp:70]: What is fine for one compiler can be "death" for another one:)

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**